### PR TITLE
fix(trap_snmp): Fix SQL injection and duplicate feature

### PR DESCRIPTION
**Pull request by callapa**
SQL queries updated to prevent SQL injection.
The duplication feature has been fixed.

### CODE REVIEW

File: www/class/centreonTraps.class.php [hunk 1]
```diff
@@ -448,64 +589,71 @@ public function insert($ret = array())
...
+        $rq .= "'" . $this->db->escape($ret["traps_output_transform"])."', ";
+        $rq .= "'" . $this->db->escape($ret['traps_advanced_treatment_default'])."', ";
+        $rq .= "'" . $this->db->escape($ret["traps_timeout"])."', ";
+        $rq .= "'" . $this->db->escape($ret["traps_customcode"])."') ";
         $this->db->query($rq);
 
-        $res = $this->db->query("SELECT MAX(traps_id) FROM traps");
-        $traps_id = $res->fetchRow();
+        $res = $this->db->query("SELECT MAX(traps_id) as trap_id FROM traps");
+        $traps_id = (int) $res->fetchRow()['trap_id'];
```
Comment (kduret) [inline www/class/centreonTraps.class.php]: not compatible with php 5.3 (centos 6)

File: www/class/centreonTraps.class.php [hunk 2]
```diff
@@ -134,183 +157,273 @@ public function testTrapExistence($oid = null)
...
-                    $query = "INSERT INTO traps_matching_properties " .
-                        "(`trap_id`,`tmo_order`,`tmo_regexp`,`tmo_string`,`tmo_status`,`severity_id`) " .
-                        "VALUES (" . $maxId['MAX(traps_id)'] . "," . $row['tmo_order'] .
-                        ",'" . $row['tmo_regexp'] . "','" . $row['tmo_string'] . "'," . $row['tmo_status'] . "," .
-                        $row['severity_id'] . ")";
-                    $this->db->query($query);
+                if (!is_null($queryValues)) {
+                    $this->db->query('INSERT INTO traps VALUES (' . $queryValues . ')');
+                    $resultMax = $this->db->query("SELECT MAX(traps_id) AS trapsId FROM traps");
+                    $newTrapId = (int) $resultMax->fetchRow()['trapsId'];
```
Comment (kduret) [inline www/class/centreonTraps.class.php]: not compatible with php 5.3 (centos 6)

File: www/class/centreonTraps.class.php (hunk 1 above)
Comment (callapa) [reply www/class/centreonTraps.class.php]: Fixed on the next commit

File: www/class/centreonTraps.class.php (hunk 2 above)
Comment (callapa) [reply www/class/centreonTraps.class.php]: Fixed on the next commit

File: www/include/configuration/configObject/traps/formTraps.php [hunk 3]
```diff
@@ -166,15 +180,32 @@ function myReplace()
...
-$form->addElement('select', 'traps_status', _("Default Status"), array(0=>_("Ok"), 1=>_("Warning"), 2=>_("Critical"), 3=>_("Unknown")), array('id' => 'trapStatus'));
+$form->addElement(
+    'select',
+    'traps_status',
+    _("Default Status"),
+    array(
+        0=>_("Ok"),
+        1=>_("Warning"),
+        2=>_("Critical"),
+        3=>_("Unknown")
```
Comment (jdelpierre) [inline www/include/configuration/configObject/traps/formTraps.php]: Add whitespace before & after the key, value 

```suggestion
         3 => _("Unknown")
```

File: www/include/configuration/configObject/traps/formTraps.php [hunk 4]
```diff
@@ -321,7 +352,17 @@ function myReplace()
...
 
-$form->addElement('select', 'traps_advanced_treatment_default', _("Advanced matching behavior"), array(0=>_("If no match, submit default status"), 1=>_("If no match, disable submit"), 2=>_("If match, disable submit")), array('id' => 'traps_advanced_treatment'));
+$form->addElement(
+    'select',
+    'traps_advanced_treatment_default',
+    _("Advanced matching behavior"),
+    array(
+        0=>_("If no match, submit default status"),
+        1=>_("If no match, disable submit"),
+        2=>_("If match, disable submit")
```
Comment (jdelpierre) [inline www/include/configuration/configObject/traps/formTraps.php]: Add whitespace before & after the key, value
```suggestion
         2 => _("If match, disable submit")
```

File: www/include/configuration/configObject/traps/formTraps.php [hunk 5]
```diff
@@ -381,19 +422,24 @@ function myReplace()
...
-if ($o == "w") {
+if ($o == TRAP_WATCH) {
     # Just watch a Command information
     if ($centreon->user->access->page($p) != 2) {
-        $form->addElement("button", "change", _("Modify"), array("onClick"=>"javascript:window.location.href='?p=".$p."&o=c&traps_id=".$traps_id."'"));
+        $form->addElement(
+            "button",
+            "change",
+            _("Modify"),
+            array("onClick"=>"javascript:window.location.href='?p=".$p."&o=c&traps_id=".$traps_id."'")
```
Comment (jdelpierre) [inline www/include/configuration/configObject/traps/formTraps.php]: Add whitespace before & after the key, value, and on the concatenation
```suggestion
             array("onClick" => "javascript:window.location.href='?p=" . $p . "&o=c&traps_id=" . $traps_id . "'")
```

File: www/include/configuration/configObject/traps/traps.php [hunk 6]
```diff
@@ -95,23 +101,27 @@
...
         require_once($path."formTraps.php");
-        break; #Watch a Trap
-    case "c":
+        break;
+    case TRAP_MODIFY:
         require_once($path."formTraps.php");
-        break; #Modify a Trap
-    case "m":
-        $trapObj->duplicate(isset($select) ? $select : array(), $dupNbr);
+        break;
```
Comment (jdelpierre) [inline www/include/configuration/configObject/traps/traps.php]: I think the 2 cases are the same requirements, so as long as there are no specific treatments for one of them I think we can group them together.
And add a space around the concatenation.

```suggestion
         case TRAP_ADD:
         case TRAP_WATCH:
         case TRAP_MODIFY:
             require_once($path . "formTraps.php");
             break;
```

Comment (callapa) [reply www/include/configuration/configObject/traps/traps.php]: Yes, you're right

**Pull request merged**